### PR TITLE
Fix column aliases handling in RelationType

### PR DESCRIPTION
**Pull request by kasiafi**
Add support for the case when there are any hidden columns before
some of the visible columns in the RelationType.
Kafka, Redis and Cassandra relations are structured this way and
before this change failed on the following query:
`SELECT a, b, c FROM (SELECT T.* FROM region T (a, b, c))`

### CODE REVIEW

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -1474,6 +1474,9 @@ public void testColumnAliases()
         assertQuery(
                 "SELECT x, T.y, z + 1 FROM (SELECT custkey, orderstatus, totalprice FROM orders) T (x, y, z)",
                 "SELECT custkey, orderstatus, totalprice + 1 FROM orders");
+
+        // wildcard from aliased table with column aliases
+        assertQuery("SELECT a, b, c FROM (SELECT T.* FROM region T (a, b, c))");
```
Comment (kokosing) [inline presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: Shouldn't we have a separate test class for these tests. I don't think there is a point to run this test for every connector. 
Are there any more tests like that besides these two queries?

This is just a thought that should probably be handled as separate PR.

@kasiafi Would you like to handle that or should I log it as github issue?

Comment (kasiafi) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: The issue with aliases applies to several connectors (Kafka, Cassandra,...). I think this is the right place to add a test.

Comment (martint) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: This is related to the discussion we had on Slack the other day, where I mentioned we need to add a way to create engine SQL tests that operate on synthetic tables. Then we could synthesize one with hidden columns just for that case.

Comment (kasiafi) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: Do we want to wait with this fix until we prepare a better place to test it?

Comment (martint) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: No, let's keep this test here. No point in forcing it to waiting behind those improvements.
